client/wfnews-war/src/main/angular/src/app/components/preview-panels/wildfire-preview/wildfire-preview.component.ts

```diff
@@ -0,0 +1,159 @@
+  styleUrls: ['./wildfire-preview.component.scss']
+})
+export class WildfirePreviewComponent implements OnDestroy{
+  public data;
+  incident: any
+  defaultZoomLevel = 13;
+  private marker: any;
+  private markerAnimation;
+
+  constructor(
+    private router: Router,
+    private appConfigService: AppConfigService,
+    private publishedIncidentService: PublishedIncidentService,
+    private cdr: ChangeDetectorRef,
+    private mapConfigService: MapConfigService
+  ) { }
+
+  ngOnDestroy(): void {
+    this.removeMarker();
+  }
+
+  snowPlowHelper = snowPlowHelper;
+  displayItemTitle = displayItemTitle;
+  getStageOfControlIcon = getStageOfControlIcon;
+  getStageOfControlLabel = getStageOfControlLabel;
+  convertToDateYear = convertToDateYear;
+  getDescription = getDescription;
+  formatNumber = formatNumber;
+  setDisplayColor = setDisplayColor;
+
+  closePanel() {
+    this.removeMarker();
+    hidePanel('desktop-preview');
+  }
+  
+  goBack(){
+    this.removeMarker();
+    showPanel('identify-panel-wrapper')
+    hidePanel('desktop-preview');
+  }
+
+  async setContent(data) {
+    this.data = data;
+    if (data.properties?.incident_number_label && data.properties?.fire_year) {
+      // identify an incident
+      try {
+        const result = await this.publishedIncidentService
+          .fetchPublishedIncident(data.properties.incident_number_label, data.properties.fire_year)
+          .toPromise();
+
+        this.incident = result;
+        console.log(result)
+        this.zoomIn();
+        this.addMarker(this.incident);
+        
+        this.cdr.markForCheck();
+
+      } catch (error) {
+        console.error('Unable to identify', error);
+
+      }
+    }
+  }
+
+  zoomIn(){
+    const long = Number(this.incident?.longitude);
+    const lat = Number(this.incident?.latitude);
+
+    if(long && lat) {
+      this.mapConfigService.getMapConfig().then(() => {
+        getActiveMap().$viewer.panToFeature(
+          window['turf'].point([long, lat]),
+          this.defaultZoomLevel,
+        );
+      });
+    }
+  }
+
+  enterFullDetail() {
+    if (this.incident?.fireYear && this.incident?.incidentNumberLabel) {
+      this.router.navigate([ResourcesRoutes.PUBLIC_INCIDENT], {
+        queryParams: {
+          fireYear: this.incident.fireYear,
+          incidentNumber: this.incident.incidentNumberLabel,
+          source: [ResourcesRoutes.ACTIVEWILDFIREMAP],
+        },
+      });
+    }
+    const url = this.appConfigService.getConfig().application.baseUrl.toString() + this.router.url.slice(1);
+    this.snowPlowHelper(url, {
+      action: 'incident_details_button_click',
+      id: this.data.incident_number_label,
+      text: 'Full Details',
+    });
+
+  }
+
+  addMarker(incident: any) {
+    if (this.marker) {
+      this.marker.remove();
+      this.marker = null;
+    }
+
+    if (this.markerAnimation) {
+      clearInterval(this.markerAnimation);
+    }
+
+    const pointerIcon = L.divIcon({
```

Comment:
there are identical function in draggable-panel, to avoid duplication, can we move them to utils index file?

Reply:
Thanks, I've updated it now